ext/standard/tests/general_functions/constant_001.phpt

```diff
@@ -0,0 +1,12 @@
+--TEST--
+Test constant() function: test access to class constant
+--FILE--
+<?php
+namespace Foo;
+
+class Bar {}
+
+var_dump(constant('Foo\Bar::class'));
```

Comment:
Because `constant('Foo\Bar::class')` is just the same as `'Foo\Bar'` that you already have to have as string, I think this need is not really real.

The current twig implementation is:
```php
function twig_constant($constant, $object = null)
{
    if (null !== $object) {
        $constant = \get_class($object).'::'.$constant;
    }

    return \constant($constant);
}
```
And by following the logic described in the first phrase, you can just make a PR to twig and have it changed to:
```php
function twig_constant($constant, $object = null)
{
    if (null !== $object) {
        if ('class' === $constant) {
            return \get_class($object);
        }
        $constant = \get_class($object).'::'.$constant;
    }

    return \constant($constant);
}
```

Reply:
Replied in https://github.com/php/php-src/pull/6763#issuecomment-794995221